test/unittest/blas3/blas3_trsm_test.cpp

```diff
@@ -0,0 +1,114 @@
+
+#include "blas_test.hpp"
+
+template<typename scalar_t>
+using combination_t = std::tuple<int, int>;
+
+template <typename T>
+void transpose(T* A, size_t m, size_t n, size_t lda) {
+  for (size_t i = 0; i < m; ++i) {
+    for (size_t j = 0; j < i; ++j) {
+      std::swap(A[j + i * lda], A[i + j * lda]);
+    }
+  }
+}
+
+template <typename T>
+void printMatrix(T* A, size_t m, size_t n, size_t lda,
+                 bool bIsColMajor = true) {
+  for (size_t i = 0; i < m; ++i) {
+    for (size_t j = 0; j < n; ++j) {
+      T Element = bIsColMajor ? A[i + j * lda] : A[j + i * lda];
+      std::cout << std::setw(4) << Element << " ";
+    }
+    std::cout << std::endl;
+  }
+}
+
+template<typename scalar_t>
+void run_test(const combination_t<scalar_t> combi) {
+  int m;
+  int n;
+  std::tie(m, n) = combi;
+
+  using data_t = utils::data_storage_t<scalar_t>;
+
+  // clang-format off
+  std::vector<data_t> A = {
+    1, 0, 0, 0,
+    2, 1, 0, 0,
+    3, 4, 1, 0,
+    5, 6, 7, 1
+  };
+  std::vector<data_t> B = {
+    1, 2, -1, -2,
+    1, 2, -1, -2,
+    1, 2, -1, -2,
+    1, 2, -1, -2
+  };
+  scalar_t alpha = scalar_t{ 1 };
+  int lda = m;
+  int ldb = m;
+  // clang-format on
```

Comment:
You can probably move `clang-format on` a big higher, I presume you just want the matrices to be nicely formatted.

Reply:
The clang-format will be removed. This was just to test a small matrix

Reply:
Fixed